docs/02_types/union.md

```diff
@@ -5,4 +5,135 @@ path: /docs/types/unions
 
 # Union types
 
-Documentation coming soon
+Union types are very similar to [interfaces](/docs/types/interfaces), but they don't have any common fields
+between the types. Here’s a union, expressed in
+[GraphQL Schema Definition Language](https://graphql.org/learn/schema/#type-language)
+(SDL):
+
+```graphql
+union MediaItem = Audio | Video | Image
+```
+
+Whenever we return a `MediaItem` in our schema, we might get an `Audio`, a
+`Video` or an `Image`. Note that members of a union type need to be concrete
+object types; you cannot create a union type out of interfaces, other unions or
+scalars.
+
+A good usecase for unions would be on a search field for example:
+
+```graphql
+searchMedia(term: "strawberry") {
+  ... on Audio {
+    duration
+  }
+  ... on Video {
+    thumbnailUrl
+  }
+  ... on Image {
+    src
+  }
+}
+```
+
+Here, the `searchMedia` field returns `[MediaItem!]!`, a list where each member
+is part of the `MediaItem` union. Since union members share no fields,
+selections are always made with [inline fragments](https://graphql.org/learn/queries/#inline-fragments).
+
+## Defining unions
+
+In Strawberry there are two ways to define a union:
+
+You can use the use the `Union` type from the `typing` module which will
+autogenerate the type name from the names of the union members:
+
+```python+schema
+from typing import Union
+import strawberry
+
+@strawberry.type
+class Audio:
+    duration: int
+
+@strawberry.type
+class Video:
+    thumbnail_url: str
+
+@strawberry.type
+class Image:
+    src: str
+
+@strawberry.type
+class Query:
+    latest_media: Union[Audio, Video, Image]
+---
+union AudioVideoImage = Audio | Video | Image
+
+type Query {
+  latestMedia: AudioVideoImage!
+}
+
+type Audio {
+  duration: Int!
+}
+
+type Video {
+  thumbnailUrl: String!
+}
+
+type Image {
+  src: String!
+}
+```
+
+Or if you need to specify a name or a description for a union you can use the
+`strawberry.union` function:
+
+```python+schema
+import strawberry
+
+MediaItem = strawberry.union("MediaItem", types=(Audio, Video, Image))
+
+@strawberry.type
+class Query:
+    latest_media: MediaItem
+---
+union MediaItem = Audio | Video | Image
+
+type Query {
+  latest_media: AudioVideoImage!
+}
+
+type Audio {
+  duration: Int!
+}
+
+type Video {
+  thumbnailUrl: String!
+}
+
+type Image {
+  src: String!
+}
+```
+
+> Note: Union types should never be instantiated directly.
```

Comment:
Probably give the alternative here?

> Union types should never be instantiated directly. Always use `strawberry.union`

Reply:
Also, why is this the case? I feel like users should be able to make the objects manually. `strawberry.union` and `StrawberryUnion` share the same signature

Reply:
Oh by this I mean that you shouldn't do this: `MediaItem(...)`. This note might be more confusing that useful, I'll remove it.